website/src/controllers/api-v3/tasks.js

```diff
@@ -102,10 +102,11 @@ api.createChallengeTasks = {
     if (challenge.leader !== user._id) throw new NotAuthorized(res.t('onlyChalLeaderEditTasks'));
 
     let tasks = await _createTasks(req, res, user, challenge);
-    res.respond(201, tasks.length === 1 ? tasks[0] : tasks);
 
     // If adding tasks to a challenge -> sync users
-    if (challenge) challenge.addTasks(tasks); // TODO catch/log
+    if (challenge) await challenge.addTasks(tasks); // TODO catch/log
```

Comment:
No, @paglias is right, this should be running in the background. Probably best to thoroughly test the behavior with unit tests, but their should be some testing in integration tests as well. Probably need to add a promisified sleep function to the helpers.

Something like:

```js
function sleep (seconds) {
  let milliseconds = seconds * 1000;

  return new Promise((resolve) => {
    setTimeout(resolve, milliseconds);
  });
}
```

Then in the tests, you can do 

```js
await sleep(2);
// fetch tasks and assert
```